src/parsec/agent/runners/lua/impl.cpp

```diff
@@ -138,25 +138,51 @@ namespace cbdc::parsec::agent::runner {
         return buf;
     }
 
+    auto lua_runner::get_stack_integer(int index) -> std::optional<int64_t> {
+        if(lua_isinteger(m_state.get(), index) != 1) {
```

Comment:
would it make sense to do this instead:

```
auto lua_runner::get_stack_integer(int index) -> std::optional<int64_t> {
         if(lua_isinteger(m_state.get(), index)) {
             return lua_tointeger(m_state.get(), index);
         }
         return std::nullopt;
     }
```

Reply:
Good catch that perhaps cleaner, though on personal preference I am going to opt to flip the branches

Reply:
Ah, apparently this is disliked by the linter! Reverting